src/neo/Wallets/NEP6/NEP6Account.cs

```diff
@@ -83,5 +85,48 @@ public bool VerifyPassword(string password)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Cache draft nep2key during wallet password changing process. Should not be called alone for a single account
+        /// </summary>
+        internal bool ChangePasswordPrepare(string password_old, string password_new)
+        {
+            if (WatchOnly) return true;
+            if (nep2key == null)
+            {
+                if (key == null)
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                try
+                {
+                    key = new KeyPair(Wallet.GetPrivateKeyFromNEP2(nep2key, password_old, wallet.Scrypt.N, wallet.Scrypt.R, wallet.Scrypt.P));
```

Comment:
This method unlock the wallet, and leave it open. I think that `key` should not be touched and use a local variable.

Reply:
Have modified, please have a look:)